portal-backend/depmap/cli_commands/db_load_commands.py

```diff
@@ -1558,6 +1558,7 @@ def _sync_metadata_to_breadbox():
         dim_type = dim_type_by_name.get(dimension_type.type_name)
         if dim_type is not None:
             metadata_dataset_id = dim_type.metadata_dataset_id
+            assert metadata_dataset_id
```

Comment:
Updating the version of the `breadbox-client` used by the `portal-backend` meant that some of the client's return types changed slightly - so I made a couple of changes like this to address new pyright errors.